### PR TITLE
Making it so there are permalinks for page

### PR DESCRIPTION
**Pull request by thalaeg**
### Summary

Updated the html to have an anchor tag which will have the get the permalink class for Terminology and Configuration headers which are contained within a plugin that is called from the basic-auth page.
I also then added the permalinks with markdown to each section also.

### Full changelog

*  Updated app/layouts/extensions.html to have a permalink
* Updated app/hub/kong-inc/basic-auth/index.md to have permalinks

### Issues resolved

Issue #904 

### Checklist:

- [x] [Commit message & atomicity](https://github.com/Kong/docs.konghq.com/blob/master/CONTRIBUTING.md#commit-atomicity) checked
- [ ] Documentation N/A
- [x] Spellchecked my updates
- [x] Ready to be merged 

This is my first time contributing to Kong and even to a really large repo. Let me know if there is anything I should do better! Thanks for giving me a shot!

### CODE REVIEW

File: app/_layouts/extension.html [hunk 1]
```diff
@@ -228,7 +228,8 @@
 
       {% if page.type == "plugin" %}
         <hr>
-        <h2 id="terminology">Terminology</h2>
+        <h2 id="terminology">
+        <a class="header-link" href="#terminology" title="Permalink"><span class="sr-only">Permalink</span><i class="fa fa-link"></i></a>Terminology</h2>
```
Comment (coopr) [inline app/_layouts/extension.html]: What problem is being solved by this change? Thanks!

Comment (thalaeg) [reply app/_layouts/extension.html]: It is adding the class for the permalink. The href in the anchor tag is adding it to the url (cause now we have an id for our h2 header) that we can now link to from other pages. In this case, now people can link to the terminology section on any HUB page!
Basically, its adding a permalink to the section header

**Comment by coopr** [comment]
This PR currently causes the Anchor link icons to _always_ appear, and to push the header text to the right:

basic_authentication_plugin___kong

What we want is what is currently available on the non /hub parts of the Docs site, which is a rollover effect, with the header text remaining left-aligned, which looks like the following, and which you can see on https://docs.konghq.com/0.14.x/configuration/ 

rollover

**Comment by thalaeg** [comment]
Okay, now the styling should be in place.. I think that may just do the trick.

**Comment by coopr** [comment]
Tests are failing - you can see it here in the PR conversation - click through to the Details of the Travis-CI built, or try `npm start` on your local copy.

**Comment by coopr** [comment]
Sorry, I'm still seeing the anchor links displayed at all times on /hub pages like http://localhost:3000/hub/kong-inc/ldap-auth-advanced/ - are you running the site locally and previewing your changes? 

ldap_authentication_advanced_plugin___kong

Sorry, I don't have any advice for you about how to implement this - I'm not familiar with how it was implemented on the rest of the site, and I don't know much about how this aspect of the site works.

**Comment by thalaeg** [comment]
No I did not install npm or docker to view it. That is part of the challenge.

I don't know that I am capable of helping with this bug without more insight on how this all works together (it's unusual because it is handled through a plugin). I don't think I can figure this out so you might want to use my branch as a starting point for someone else cause I think I am close. Thanks for your checks thus far and helping to make a cool product!

**Comment by thalaeg** [comment]
Shoot, I was trying to commit that textual errors into a different pull request.

**Pull request closed without merging**